RHEL6_7/packages/RemovedPackages/noauto_postupgrade.d/native_rpm_dep_fix.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+#RHSIGNED_PKGS="/var/cache/preupgrade/common/rpm_rhsigned.log"
+RHSIGNED_PKGS="rpm_rhsigned.log"
+kept_broken_rpms="kept_broken_rpms"
+VERBOSE=0
+RM_CUSTOM_RPM=0
+FAIL=0
+
+log_error()   { echo >&2 "Error:   $@"; }
+log_warning() { echo >&2 "Warning: $@"; }
+log_info()    { echo >&2 "Info:    $@"; }
+
+log_info_verbose() {
+    [ 0"$VERBOSE" -eq 1 ] && log_info "$@"
+}
+
+get_broken_deps_list() {
+    rpm -Va --nofiles \
+        | grep -i "^Unsatisfied" \
+        | sed "s/^Unsatisfied dependencies for //" \
+        | sed "s/:$//" \
+        | sort \
+        | uniq
+}
+
+is_dist_native() {
+    grep -qE "^$1([[:space:]]|$)" "$RHSIGNED_PKGS"
+}
+
+rm_broken_old_rpms() {
+    #
+    # Remove old RPMs with broken dependencies.
+    #
+    # Accepts one parameter, which says whether will be removed only native
+    # old RPMs with broken dependencies (default) or should be removed even
+    # non-native old RPMs with broken dependencies (when param=1)
+    #
+    # Returns 0 when no RPM has been removed. Otherwise returns 1.
+    #
+    local rm_custom_enabled=0
+    [ "$1" -eq 1 ] && rm_custom_enabled=1
+
+    get_broken_deps_list | grep "\.el6" > rpms_broken
+    local REMOVED_FLAG=0
+    local line=""
+    while IFS= read -r line || [ -n "$line" ]; do
+        NAME=$(rpm -q --qf '%{NAME}' $line)
+        is_dist_native "$NAME" || [ $rm_custom_enabled -eq 1 ] || {
+            # it's not dist native package -> skip it
+            echo "    $NAME" >> "$kept_broken_rpms"
+            log_info_verbose "Skip the 3rd party of custom $line RPM."
+            continue
+        }
+
+        # remove broken rpm
+        log_info_verbose "Removing the $line RPM."
+        rpm -e --nodeps "$line" || {
+            rpm -q "$line" >/dev/null 2>&1 || {
+                log_error "The $line RPM has not been removed."
+                FAIL=1
+            }
+            continue
+        }
+        REMOVED_FLAG=1
+    done < rpms_broken
+    return $REMOVED_FLAG
+}
+
+
+##################### MAIN ###########################
+
+# process params from cmdln
+while [[ -n "$1" ]]; do
+    case $1 in
+        -a | --rm-all)
+            RM_CUSTOM_RPM=1
+            ;;
+        -v | --verbose)
+            VERBOSE=1
+            ;;
+        -h | --help)
+            echo "USAGE: native_rpm_dep_fix [-a|--rm-all] [-v|--verbose] [-h|--help]"
```

Comment:
`native_rpm_dep_fix` is not true anymore.  You could use `${0##*/}` here, I think.

Reply:
right! will be fixed in the next commit.